src/librbd/cache/WriteLogImageDispatch.cc

```diff
@@ -198,26 +197,26 @@ bool ImageDispatch<I>::flush(
 
   aio_comp->set_request_count(1);
   io::C_AioRequest *req_comp = new io::C_AioRequest(aio_comp);
-  m_image_cache->aio_flush(flush_source, req_comp);
+  m_image_cache->flush(flush_source, req_comp);
 
   return true;
 }
 
 template <typename I>
-bool ImageDispatch<I>::list_snaps(
+bool WriteLogImageDispatch<I>::list_snaps(
     io::AioCompletion* aio_comp, io::Extents&& image_extents,
     io::SnapIds&& snap_ids,
     int list_snaps_flags, io::SnapshotDelta* snapshot_delta,
     const ZTracer::Trace &parent_trace, uint64_t tid,
     std::atomic<uint32_t>* image_dispatch_flags,
     io::DispatchResult* dispatch_result, Context** on_finish,
     Context* on_dispatched) {
-  ceph_abort();
+  return false;
```

Comment:
It needs to call list_snaps in TestMockDeepCopyObjectCopyRequest::Write in which RWL may be enabled for the source volume.

Reply:
Does this fix the test failure? When I checked with this modification, the UT is still failing

Reply:
That's failing for me because `ReadResult::set_image_extents` is never invoked on the PWL path so it fails to reassemble a sparse read:

```
2020-10-13T10:21:27.610-0400 7f9f9b7fe700 20 librbd::io::ReadResult: 0x7f9f9b7f9bb0 operator(): image_extents=[], buffer_extent_map={0=3704018}
/home/jdillaman/ceph_wip/src/librbd/io/ReadResult.cc: In function 'void librbd::io::ReadResult::AssembleResultVisitor::operator()(librbd::io::ReadResult::SparseBufferlist&) const' thread 7f9f9b7fe700 time 2020-10-13T10:21:27.612061-0400
/home/jdillaman/ceph_wip/src/librbd/io/ReadResult.cc: 140: FAILED ceph_assert(buffer_offset == buffer_extents_length)
```

Reply:
... and `TestMockCacheReplicatedWriteLog.init_state_write` consistently fails as well but unrelated to this PR.

Reply:
Thank you for pointing it out. Let me investigate.

Reply:
https://github.com/ceph/ceph/pull/37669 is to fix the failure in TestMockCacheReplicatedWriteLog.init_state_write.